src/logics/syncswap/logic.swap-token.test.ts

```diff
@@ -0,0 +1,119 @@
+        const tokenList = await logic.getTokenList();
+        expect(tokenList).to.have.lengthOf.above(0);
+      });
+    });
+  });
+
+  context('Test build', function () {
+    const chainId = common.ChainId.zksync;
+    const logic = new SwapTokenLogic(chainId);
+    const routerAddress = getContractAddress(chainId, 'Router');
+    const iface = Router__factory.createInterface();
+
+    const testCases: LogicTestCase<SwapTokenLogicFields, SwapTokenLogicOptions>[] = [
+      {
+        fields: {
+          input: new common.TokenAmount(zksyncTokens.ETH, '1'),
+          output: new common.TokenAmount(zksyncTokens.USDC, '0'),
+          paths: [
+            {
+              steps: [
+                {
+                  pool: '0x80115c708E12eDd42E504c1cD52Aea96C547c05c',
+                  tokenIn: '0x5AEa5775959fBC2557Cc8789bC1bf90A239D9a91',
+                },
+              ],
+              tokenIn: '0x5AEa5775959fBC2557Cc8789bC1bf90A239D9a91',
+              amountIn: '1000000000000000000',
+            },
+          ],
+          slippage: 500,
+        },
+        options: { account: '0xaAaAaAaaAaAaAaaAaAAAAAAAAaaaAaAaAaaAaaAa' },
+      },
+      {
+        fields: {
+          input: new common.TokenAmount(zksyncTokens.USDC, '1'),
+          output: new common.TokenAmount(zksyncTokens.ETH, '0'),
+          paths: [
+            {
+              steps: [
+                {
+                  pool: '0x80115c708E12eDd42E504c1cD52Aea96C547c05c',
+                  tokenIn: '0x3355df6D4c9C3035724Fd0e3914dE96A5a83aaf4',
+                },
+              ],
+              tokenIn: '0x3355df6D4c9C3035724Fd0e3914dE96A5a83aaf4',
+              amountIn: '1000000',
+            },
+          ],
+          slippage: 500,
+        },
+        options: { account: '0xaAaAaAaaAaAaAaaAaAAAAAAAAaaaAaAaAaaAaaAa' },
+      },
+      {
+        fields: {
+          input: new common.TokenAmount(zksyncTokens.USDC, '1'),
+          output: new common.TokenAmount(zksyncTokens.WBTC, '0'),
+          paths: [
+            {
+              steps: [
+                {
+                  pool: '0x80115c708E12eDd42E504c1cD52Aea96C547c05c',
+                  tokenIn: '0x3355df6D4c9C3035724Fd0e3914dE96A5a83aaf4',
+                },
+                {
+                  pool: '0xb3479139e07568BA954C8a14D5a8B3466e35533d',
+                  tokenIn: '0x5AEa5775959fBC2557Cc8789bC1bf90A239D9a91',
+                },
+              ],
+              tokenIn: '0x3355df6D4c9C3035724Fd0e3914dE96A5a83aaf4',
+              amountIn: '1000000',
+            },
+          ],
+          slippage: 500,
+        },
+        options: { account: '0xaAaAaAaaAaAaAaaAaAAAAAAAAaaaAaAaAaaAaaAa' },
+      },
+    ];
+
+    testCases.forEach(({ fields, options }) => {
+      it(`${fields.input.token.symbol} to ${fields.output.token.symbol}`, async function () {
+        const routerLogic = await logic.build(fields, options);
+        const sig = routerLogic.data.substring(0, 10);
```

Comment:
could explain why the length is 10 in the comments.

Reply:
[Oral Discussion]